client/state/notices/test/middleware.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { createStore } from 'redux';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import noticesMiddleware, { OBSERVERS } from '../middleware';
+import { useSandbox } from 'test/helpers/use-sinon';
+import { successNotice } from 'state/notices/actions';
+import {
+	NOTICE_CREATE,
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+describe( 'middleware', () => {
+	let state = null,
+		store;
+
+	useSandbox( ( sandbox ) => {
+		store = createStore( noop );
+		sandbox.stub( store, 'getState', () => state );
+		sandbox.spy( store, 'dispatch' );
+	} );
+
+	beforeEach( () => {
+		state = null;
+	} );
+
+	describe( 'noticesMiddleware()', () => {
+		function dummyNoticeObserver( action, dispatch, getState ) {
+			dispatch( successNotice( `${ getState() } ${ action.target }` ) );
+		}
+
+		before( () => {
+			OBSERVERS.DUMMY_TYPE = dummyNoticeObserver;
+		} );
+
+		after( () => {
+			delete OBSERVERS.DUMMY_TYPE;
+		} );
+
+		it( 'should trigger the observer corresponding to the dispatched action type', () => {
+			state = 'Hello';
+			noticesMiddleware( store )( noop )( { type: 'DUMMY_TYPE', target: 'World' } );
+
+			expect( store.dispatch ).to.have.been.calledWithMatch( {
+				type: NOTICE_CREATE,
+				notice: {
+					text: 'Hello World'
+				}
+			} );
```

Comment:
I haven't seen this being done. It would be great to have a write-up on testing dispatched actions like this in the Calypso digest

cc: @mtias